problem/tools/settings/unittest.cc

```diff
@@ -0,0 +1,133 @@
+}
+// Horizontal Streak
+TEST(TTT, HorizontalStreak){
+  const std::vector<std::vector<std::string>> horStreak1 = {{"X", "X", "X"}, {"_", "_", "_"}, {"_", "_", "_"}};
+  const std::vector<std::vector<std::string>> horStreak2 = {{"_", "_", "_"}, {"O", "O", "O"}, {"_", "_", "_"}};
+  const std::vector<std::vector<std::string>> horStreak3 = {{"X", "X", "O"}, {"O", "X", "X"}, {"O", "O", "O"}};
+  ASSERT_EQ(PlayerWonCheck(horStreak1, 1), true) << "If it is currently player 1, it should return true when there is a \"horizontal streak\" or when all the values of gameBoard[0][value] = \"X\"";
+  ASSERT_EQ(PlayerWonCheck(horStreak2, 2), true) << "If it is currently player 2, it should return true when there is a \"horizontal streak\" or when all the values of gameBoard[1][value] = \"O\"";
+  ASSERT_EQ(PlayerWonCheck(horStreak3, 2), true) << "If it is currently player 2, it should return true when there is a \"horizontal streak\" or when all the values of gameBoard[2][value] = \"O\"";
+}
+// Vertical Streak
+TEST(TTT, VerticalStreak){
+  const std::vector<std::vector<std::string>> verStreak1 = {{"X", "_", "_"}, {"X", "_", "_"}, {"X", "_", "_"}};
+  const std::vector<std::vector<std::string>> verStreak2 = {{"_", "O", "_"}, {"_", "O", "_"}, {"_", "O", "_"}};
+  const std::vector<std::vector<std::string>> verStreak3 = {{"O", "O", "X"}, {"O", "X", "X"}, {"X", "O", "X"}};
+  ASSERT_EQ(PlayerWonCheck(verStreak1, 1), true) << "If it is currently player 1, it should return true when there is a \"vertical streak\" or when all the values of gameBoard[value][0] = \"X\"";
+  ASSERT_EQ(PlayerWonCheck(verStreak2, 2), true) << "If it is currently player 2, it should return true when there is a \"vertical streak\" or when all the values of gameBoard[value][1] = \"O\"";
+  ASSERT_EQ(PlayerWonCheck(verStreak3, 1), true) << "If it is currently player 1, it should return true when there is a \"vertical streak\" or when all the values of gameBoard[value][2] = \"X\"";
+}
+// Diagonal Left
+TEST(TTT, LeftDiagonalStreak){
+  const std::vector<std::vector<std::string>> lefDStreak1 = {{"X", "_", "_"}, {"_", "X", "_"}, {"_", "_", "X"}};
+  const std::vector<std::vector<std::string>> lefDStreak2 = {{"O", "X", "_"}, {"_", "O", "X"}, {"X", "O", "O"}};
+  ASSERT_EQ(PlayerWonCheck(lefDStreak1, 1), true) << "If it is currently player 1, it should return true when there is a \"left to right diagonal streak\" or when [0][0], [1][1], [2][2] = \"X\"";
+  ASSERT_EQ(PlayerWonCheck(lefDStreak2, 2), true) << "If it is currently player 2, it should return true when there is a \"left to right diagonal streak\" or when [0][0], [1][1], [2][2] = \"O\"";
+}
+// Diagonal Right
+TEST(TTT, RightDiagonalStreak){
+  const std::vector<std::vector<std::string>> rightDStreak1 = {{"_", "_", "X"}, {"_", "X", "_"}, {"X", "_", "_"}};
+  const std::vector<std::vector<std::string>> rightDStreak2 = {{"X", "X", "O"}, {"_", "O", "X"}, {"O", "O", "_"}};
```

Comment:
rewrite hint to say: "or when the top right and its opposite corner makes a "diagonal line"